### DISCUSSION TITLE
Click on deposit or reserved

### DISCUSSION THREAD

**Discussion by edsonblwolf** [netbox-community/netbox#13404] [Help Wanted!]
### NetBox version

v3.5.7

### Python version

3.11

### Steps to Reproduce

When I click on devices or reservations it goes to localhost:
http://localhost:8001/dcim/devices/1/

### Expected Behavior

http://192.168.2.12/dcim/devices/1/

### Observed Behavior

http://localhost:8001/dcim/devices/1/

[https://imgur.com/WwnYRjs.png](url)

**Comment by candlerb** [comment]
Your front-end HTTP proxy config (Apache or Nginx) is incomplete - it's not proxying the hostname and port which the client supplied.

There are working configs you can copy here: https://github.com/netbox-community/netbox/tree/master/contrib

With Apache you're likely missing the `ProxyPreserveHost On` setting.  I don't know about Nginx.

**Comment by edsonblwolf** [comment]
It's just when I'm in the rack, the system is normal.

**Reply by candlerb** [reply]
That's correct.

Most hyperlinks that Netbox generates are relative, so don't care what the hostname is. But in a few places, such as rack elevations, the hyperlinks are absolute: in those cases, the Netbox (gunicorn) server needs to know what hostname you were accessing, in order to generate the correct hostnames in those links.

(I believe this is so that if you copy one of these images and use it somewhere else, the links will continue to work)

**Comment by edsonblwolf** [comment]
If I put the IP works, but the images do not appear. Why is localhost defined.

**Reply by candlerb** [reply]
> If I put the IP works, but the images do not appear.

Most likely the image links point to `http://localhost/` as well. You can use your browser's console to inspect the HTML.

> Why is localhost defined.

I assume you mean "why does Netbox generate links to localhost"?

It's because your front-end proxy (Apache or Nginx) is configured to forward incoming requests to localhost:8001, and if you don't configure it to pass on the hostname from the original request, then Netbox thinks your browser asked for "localhost:8001", so it builds absolute links starting with `http://localhost:8001/`

The problem you have is one I've seen several times before, and the solution is what I already gave you.
